app-directory/share-state/app/page.tsx

```diff
@@ -0,0 +1,179 @@
+import { Page, Text, Link, Code, Snippet } from '@vercel/examples-ui'
+
+export default function Home() {
+  return (
+    <Page className="flex flex-col gap-12">
+      <section className="flex flex-col gap-6">
+        <Text variant="h1">Share state within app dir</Text>
+        <Text>
+          Sometimes you need to share state between pages or layouts within your
+          app. It may be complex as your pages fetch data from the server or you
+          need to track state on your layout from user interactions within
+          nested routes.
+        </Text>
+      </section>
+
+      <section className="flex flex-col gap-3">
+        <Text variant="h2">Client layouts and server pages</Text>
+        <Text>
+          Althought client components can&apos;t include server components, we
+          can have a client layout rendering a page from the server. This allows
+          us to create a state or context to share data across our application.
+        </Text>
+      </section>
+
+      <section className="flex flex-col gap-3">
+        <Text variant="h2">Creating the files</Text>
+        <Text>
+          We will create a root layout for our app that will show the counter
+          state, a context to share state, an index and a nested page that
+          displays the counter component and a counter client component to
+          interact with our state.
+        </Text>
+        <pre className="border border-accents-2 rounded-md bg-accents-1 overflow-x-auto p-6">{`|/app
+|__/layout.js (client)
+|__/context.js (client)
+|__/counter.js (client)
+|__/page.js (server)
+|__/nested
+|____/page.js (server)
+`}</pre>
+        <Text>
+          Lets create a simple context that stores a <Code>counter</Code> value
+          and exports a <Code>useCounter</Code> hook to interact with that
+          value.
+        </Text>
+        <Snippet>
+          {`import { createContext, useContext, useState } from "react";
+
+const CounterContext = createContext({});
```

Comment:
Would it be possible to add a comment here as to why Context API should be used instead of just using `useState` hook?